state/state.go

```diff
@@ -0,0 +1,91 @@
+package state
+
+import (
+	"log"
+	"sync"
+
+	"github.com/NordSecurity/nordvpn-linux/events"
+	"github.com/NordSecurity/nordvpn-linux/internal"
+)
+
+type Event int
+
+const (
+	VPNConnected Event = iota
+	VPNDisconnected
+)
+
+type subscriber struct {
+	stateChan chan<- Event
+	stopChan  <-chan struct{}
+}
+
+func newSubscriber(stateChan chan<- Event, stopChan <-chan struct{}) subscriber {
+	return subscriber{
+		stateChan: stateChan,
+		stopChan:  stopChan,
+	}
+}
+
+type subscriberSet map[subscriber]bool
+
+type StatePublisher struct {
+	mu          sync.Mutex
+	subscribers subscriberSet
```

Comment:
why does it need to be a set/map, because there is no lookup or any other functionality specific to maps used?